crates/ruff/src/rules/flake8_pyi/rules/mod.rs

```diff
@@ -69,3 +70,24 @@ mod unrecognized_platform;
 mod unrecognized_version_info;
 mod unsupported_method_call_on_all;
 mod unused_private_type_definition;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum TypingModule {
+    Typing,
+    TypingExtensions,
+}
+
+impl TypingModule {
```

Comment:
I wonder if it would be more useful to generalize this into a generic backport import helper. 

IE. something that would automatically downgrade typing.TypeAlias. Alternatively, I wonder if you could use the pyupgrade import rules logic here and always apply that rule, but only to this specific import. It would either be a NOOP or automatically fix the import if the import was moved to the typing module.

That also has the added benefit of only having to maintain the version when a new import was introduced in one place (the pyupgrade rules).

Reply:
The logic could even be folded into the helper which inserts the import if you do it pyupgrade way.

Reply:
```
Note that, in some cases, it may be preferable to continue importing members from typing_extensions even after they're added to the Python standard library, as typing_extensions can backport bugfixes and optimizations from later Python versions. This rule thus avoids flagging imports from typing_extensions in such cases.
```
Seems like more reason to use the helper from this rule here.

Reply:
Nice idea, I will look into it separately. There are a few other places where this would be useful, like `PYI019` which suggests `typing.Self`.

Reply:
Yup agreed. (I did check if that applied to `TypeAlias` and it does not.)

Reply:
Adding a TODO, I may look into it tonight.